contract/src/main.rs

```diff
@@ -1271,206 +1277,353 @@ pub extern "C" fn set_token_metadata() {
+    // installation. Meant to be called by the managing account (INSTALLER) post
+    // installation if a variable needs to be changed. Each parameter of the entrypoint
+    // should only be passed if that variable is changed.
+    // For instance if the allow_minting variable is being changed and nothing else
+    // the managing account would send the new allow_minting value as the only argument.
+    // If no arguments are provided it is essentially a no-operation, however there
+    // is still a gas cost.
+    // By switching allow_minting to false we pause minting.
+    let set_variables = EntryPoint::new(
+        ENTRY_POINT_SET_VARIABLES,
+        vec![
+            Parameter::new(ARG_ALLOW_MINTING, CLType::Bool),
+            Parameter::new(
+                ARG_CONTRACT_WHITELIST,
+                CLType::List(Box::new(CLType::ByteArray(32u32))),
+            ),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    );
+
+    // This entrypoint mints a new token with provided metadata.
+    // Meant to be called post installation.
+    // Reverts with MintingIsPaused error if allow_minting is false.
+    // When a token is minted the calling account is listed as its owner and the token is
+    // automatically assigned an U64 ID equal to the current number_of_minted_tokens.
+    // Before minting the token the entrypoint checks if number_of_minted_tokens
+    // exceed the total_token_supply. If so, it reverts the minting with an error
+    // TokenSupplyDepleted. The mint entrypoint also checks whether the calling account
+    // is the managing account (the installer) If not, and if public_minting is set to
+    // false, it reverts with the error InvalidAccount. The newly minted token is
+    // automatically assigned a U64 ID equal to the current number_of_minted_tokens. The
+    // account is listed as the token owner, as well as added to the accounts list of owned
+    // tokens. After minting is successful the number_of_minted_tokens is incremented by
+    // one.
+    let mint = EntryPoint::new(
+        ENTRY_POINT_MINT,
+        vec![
+            Parameter::new(ARG_TOKEN_OWNER, CLType::Key),
+            Parameter::new(ARG_TOKEN_META_DATA, CLType::String),
+        ],
+        CLType::Tuple3([
+            Box::new(CLType::String),
+            Box::new(CLType::Key),
+            Box::new(CLType::String),
+        ]),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    );
+
+    // This entrypoint burns the token with provided token_id argument, after which it is no
+    // longer possible to transfer it.
+    // Looks up the owner of the supplied token_id arg. If caller is not owner we revert with
+    // error InvalidTokenOwner. If token id is invalid (e.g. out of bounds) it reverts
+    // with error  InvalidTokenID. If token is listed as already burnt we revert with
```

Comment:
extra whitespace, also seems like two entry points comments shoved together

Reply:
I think the comment is for this entry point alone